src/spikeinterface/core/template.py

```diff
@@ -294,9 +294,9 @@ def from_zarr_group(cls, zarr_group: "zarr.Group") -> "Templates":
         the `add_templates_to_zarr_group` method.
 
         """
-        templates_array = zarr_group["templates_array"]
-        channel_ids = zarr_group["channel_ids"]
-        unit_ids = zarr_group["unit_ids"]
+        templates_array = zarr_group["templates_array"][:]
```

Comment:
I think there is something to watch out when having this behavior by default. If we are serializing something in multiple places then we will keep making the full read even if the full function is not used. This might be important for reducing the number of calls to S3.

On the other hand this might be better that someone doing multiple requests without the copy...

Myabe instead of going all the way manual as in https://github.com/SpikeInterface/spikeinterface/pull/2853/files we can have a method called get_templates that does this when it is called?

Just thinking aloud.

Reply:
@h-mayorquin  I am agree but here this is a class method and we really want a in memory loading no ?